Frontend/src/pages/WeightsPage.jsx

```diff
@@ -3,250 +3,319 @@ import "../styles/classforge.css";
+              <li><Link to="/ai-query">AI Query</Link></li>
+            </ul>
+          </nav>
+        </div>
+      </header>
+
+      {/* Dataset Upload */}
+      <section id="dataset-upload">
+        <div className="container">
+          <h2>Upload Dataset</h2>
+          <form onSubmit={handleDatasetUpload}>
+            <div className="form-group">
+              <label htmlFor="collectionName">Collection Name:</label>
+              <input
+                type="text"
+                id="collectionName"
+                value={collectionName}
+                onChange={e => setCollectionName(e.target.value)}
+                placeholder="Enter collection name"
+                required
+              />
             </div>
-          </div>
-        </section>
+            <div className="form-group">
+              <label htmlFor="datasetFile">Select Dataset File (CSV):</label>
+              <input
+                type="file"
+                id="datasetFile"
+                accept=".csv"
+                onChange={handleFileChange}
+                required
+              />
+            </div>
+            <button type="submit" disabled={isUploading}>
+              {isUploading ? "Uploading..." : "Upload Dataset"}
+            </button>
+          </form>
+          {uploadMessage && (
+            <div className={uploadMessage.startsWith("❌") ? "error-message" : "success-message"}>
+              {uploadMessage}
+            </div>
+          )}
+        </div>
+      </section>
 
-        {/* Footer */}
-        <footer>
-          <div className="container">
-            <p>&copy; 2025 ClassForge. All rights reserved.</p>
+      {/* Allocation */}
+      <section id="allocation">
+        <div className="container">
+          <h2>Classroom Allocation</h2>
+          <form id="allocationForm" onSubmit={handleAllocation}>
+            {/* Primary 0–100 sliders */}
+            <div className="form-group">
+              <label htmlFor="academicWeight">Academic Performance Weight:</label>
+              <input
+                type="range"
+                id="academicWeight"
+                min="0"
+                max="100"
+                value={academicWeight}
+                onChange={e => setAcademicWeight(Number(e.target.value))}
+              />
+              <span>{academicWeight}%</span>
+            </div>
+            <div className="form-group">
+              <label htmlFor="socialWeight">Social Integration Weight:</label>
+              <input
+                type="range"
+                id="socialWeight"
+                min="0"
+                max="100"
+                value={socialWeight}
+                onChange={e => setSocialWeight(Number(e.target.value))}
+              />
+              <span>{socialWeight}%</span>
+            </div>
+            <div className="form-group">
+              <label htmlFor="wellBeingWeight">Well-Being Weight:</label>
```

Comment:
Same as what I was saying earlier, could I double check what wellbeing effects? Or is this a GA thing?